indexers/keyvalue/MongoDBIndexer/__init__.py

```diff
@@ -46,13 +44,15 @@ def get_create_handler(self):
     def get_query_handler(self):
         return self.handler
     
-    def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
-        mongo_docs = [{'_id': i, 'values': j} for i, j in zip(keys, values)]
-        
+    def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs) -> None:
+        total_inserted_ids = []
         with self.write_handler as mongo_handler:
-            inserted_ids = mongo_handler.insert(documents=mongo_docs)
+            for i, j in zip(keys, values):
+                doc = {'_id': i, 'values': j}
+                inserted_ids = mongo_handler.insert(documents=[doc])
+                total_inserted_ids.extend(inserted_ids)
         
-        if inserted_ids and len(inserted_ids) != len(keys):
+        if total_inserted_ids and len(total_inserted_ids) != len(list(keys)):
```

Comment:
why is it needed the transform to `len(list(keys))`?

Reply:
`Expected type 'Sized', got 'Iterator[int]' instead`. Just to calm down PyCharm